rs/boundary_node/ic_boundary/src/metrics.rs

```diff
@@ -651,18 +652,33 @@ pub async fn metrics_middleware(
             .observe(response_size as f64);
 
         // Anonymization
-        let s = anonymization_salt.load();
+        let salt = anonymization_salt.load();
 
-        let hash_fn = |v: &str| -> String {
-            if s.is_none() {
-                return "N/A".to_string();
-            }
+        let hash_fn = |input: &str| -> String {
+            let err_str = "N/A".to_string();
+
+            let salt = match salt.as_ref() {
+                Some(s) if s.len() == SALT_SIZE => s,
+                _ => return err_str,
+            };
+
+            let mut hasher = {
+                // Convert first 8 bytes to u64, return N/A if it fails
+                let key0 = match salt[0..8].try_into() {
+                    Ok(bytes) => u64::from_le_bytes(bytes),
+                    Err(_) => return err_str,
+                };
+                // Convert next 8 bytes to u64, return N/A if it fails
+                let key1 = match salt[8..16].try_into() {
+                    Ok(bytes) => u64::from_le_bytes(bytes),
+                    Err(_) => return err_str,
+                };
+                SipHasher::new_with_keys(key0, key1)
+            };
 
-            let mut h = DefaultHasher::new();
-            v.hash(&mut h);
-            s.hash(&mut h);
+            input.hash(&mut hasher);
```

Comment:
This discards entirely the last half of the salt. I'd suggest either just using only a 16 byte salt (but this may require changes elsewhere, AIUI?) or providing the last half of the salt here along with the value being hashed.

Reply:
Let's proceed with the second option, as 32-byte salt is already generate by the canister.
I'd use the entire salt for hashing and avoid slicing, it won't hurt to use all 32 bytes i guess:
```
// Hash the entire 32-byte salt to ensure all bytes are used
salt.hash(&mut hasher);
input.hash(&mut hasher);
```
@randombit have a final look plz and approve if all is well

Reply:
Right now you pass in the full salt and not just the "last half" as suggested.

Reply:
Yes, i wanted to avoid slicing, if using the whole salt won't hurt.